### DISCUSSION TITLE
HmIP-WRCC2: Tastendruck als Event

### DISCUSSION THREAD

**Discussion by carusoe** [OpenCCU/OpenCCU#2133] [🇩🇪 Fragen&Antworten] [answered]
Hallo,
ich habe mir den Homematic IP Schalter HmIP-WRCC2 zugelegt, jedoch ohne vorher zu schauen, ob man ein Licht über den Home Assistant schalten kann... :-/

Hier ist die Situation: Ich habe auf einem Raspberry mit dem Modul RPI-RF-MOD Home Assistant mit Raspberrymatic als Plugin installiert. In Home Assistant habe ich eine Lifx Birne und wollte jetzt den HmIP-WRCC2-Schalter dazu nutzen, um die Lifx zu schalten. Also Taste 1 auf dem HmIP-WRCC2 -> Lifx an und Taste 2 -> Lifx aus. (ich weiß man kann es auch intelligenter machen - nur damit Ihr meine Idee versteht!)

Jetzt habe ich gelesen, dass aufgrund einer Einschränkung der EQ3-Integration(?) die Events der Tastendrucke nicht an Home Assistant weitergegeben werden. Zumindest sehe ich in Home Assistant keine entsprechenden Events

Daher meine Fragen:
1. Ist es wirklich so, dass die EQ3-Integration noch nicht so weit ist?
2. Gibt es nicht einen anderen Weg? (Ich habe irgendwas gelesen, dass es über den IO-Broker gehen soll...Anleitung?)
3. Wann soll eigentlich Matter bei Home Assistant und Raspberrymatic/EQ3 einziehen?

Danke schon einmal für Eure Hilfe!

**Comment by Baxxy13** [comment, marked as answer]
Zu 1: 
Das hat mit der Integration nix zu tun. Die Zentrale (CCUx/RM) sendet per default keine IP-Tasterevents an RPC-Clients (Integration, ioBroker, NodeRed usw.)

Zu 2: 
Der alte (und auf CCUx immer noch nötige) Weg war/ist ein Dummy-Programm auf der Zentrale anzulegen welches auf den Tastendruck reagiert. 
In RaspberryMatic kann man die Events inzwischen auch ohne Dummy-Programm aktivieren/deaktivieren indem man in "> Einstellungen > Geräte" einen IP-Tasterkanal anklickt und dann das "Tasterevent an Zentrale" aktiviert/deaktiviert

**Reply by carusoe** [reply]
Hallo Baxxy13,

Danke für den Tipp. Der hat mein Problem gelöst. :-) Jetzt muss ich nur noch herausfinden wieso manchmal die Birne reagiert und warum manchmal nicht. Wahrscheinlich muss ich in den Einstellungen des Schalters noch ein paar Parameter anpassen...

**Reply by Baxxy13** [reply]
Wenn du mit "Schalter" den HmIP-Taster meinst dann muss man da i.d.R. nichts anpassen. 
Die "Mindestdauer für langen Tastendruck" könnte man etwas hochsetzen damit "langsamere Menschen" den nicht versehentlich auslösen.

Kommt das Event in der Zentrale (RM / CCUx, Tasterkanal protokollieren) und auch im HA (Entwicklerwerkzeuge --> Ereignisse --> hören auf Ereignis: `homematic.keypress` ) an, dann ist von dieser Seite alles in Ordnung.
